src/braket/ocean_plugin/braket_sampler.py

```diff
@@ -308,28 +356,53 @@ def sample_qubo_quantum_task(self, Q: Dict[Tuple[int, int], int], **kwargs) -> Q
 
         Examples:
             This example submits a two-variable QUBO mapped directly to qubits
-            0 and 4 on a sampler
+            0 and 4 on a sampler on the D-Wave 2000Q device.
 
             >>> from braket.ocean_plugin import BraketSampler
-            >>> sampler = BraketSampler(s3_destination_folder, "device_arn_1")
+            >>> device_arn_1 = "arn:aws:braket:::device/qpu/d-wave/DW_2000Q_6"
+            >>> sampler = BraketSampler(s3_destination_folder, device_arn_1)
             >>> Q = {(0, 0): -1, (4, 4): -1, (0, 4): 2}
             >>> task = sampler.sample_qubo_quantum_task(Q, resultFormat="HISTOGRAM", shots=100)
             >>> sampleset = BraketSampler.get_task_sample_set(task)
             >>> for sample in sampleset.samples():
             ...    print(sample)
             ...
+            {0: 0, 4: 1}
             {0: 1, 4: 0}
+
+            This example submits a two-variable QUBO mapped directly to qubits
+            30 and 31 on a sampler on the D-Wave Advantage4 device.
+
+            >>> from braket.ocean_plugin import BraketSampler
+            >>> device_arn_1 = "arn:aws:braket:::device/qpu/d-wave/Advantage_system4"
+            >>> sampler = BraketSampler(s3_destination_folder, device_arn_1)
+            >>> Q = {(30, 30): -1, (31, 31): -1, (30, 31): 2}
+            >>> task = sampler.sample_qubo_quantum_task(Q, resultFormat="HISTOGRAM", shots=100)
+            >>> sampleset = BraketSampler.get_task_sample_set(task)
+            >>> for sample in sampleset.samples():
+            ...    print(sample)
+            ...
+            {30: 0, 31: 1}
+            {30: 1, 31: 0}
         """
         solver_kwargs = self._process_solver_kwargs(**kwargs)
 
         sorted_edges = frozenset((u, v) if u < v else (v, u) for u, v in Q)
-        if not all(
-            u in self._access_optimized_nodelist()
-            if u == v
-            else v in self._access_optimized_edgelist().get(u, ())
-            for u, v in sorted_edges
-        ):
-            raise BinaryQuadraticModelStructureError("Problem graph incompatible with solver.")
+        print(self._access_optimized_edgelist())
+        for u, v in sorted_edges:
+            if u not in self._access_optimized_nodelist():
+                raise BinaryQuadraticModelStructureError(
+                    "Problem graph incompatible with solver. Qubits are not in the device's qubit "
+                    "set."
+                )
+            if v not in self._access_optimized_edgelist().get(u, ()) and u != v:
+                raise BinaryQuadraticModelStructureError(
+                    "Problem graph incompatible with solver. Nodes "
+                    + str(u)
+                    + " and "
+                    + str(v)
+                    + " are not connected."
```

Comment:
I think "Solver nodes $u and $v are not connected" would clarify that the issue is the `Problem` has nodes that are connected that should not be.